Materials/Materials/Program.cs

```diff
@@ -13,135 +13,38 @@
 	options.AddPolicy("AllowSpecificOrigin",
 		builder =>
 		{
-			builder.WithOrigins("http://localhost:3000") // Replace with the actual origin of your frontend
+			builder.WithOrigins("https://localhost:3000") // Replace with the actual origin of your frontend
 				.AllowAnyHeader()
 				.AllowAnyMethod();
 		});
 });
 
-builder.Services.AddMvc(options =>
-{
-	options.Filters.Add<CustomExceptionFilter>();
-});
-
 var app = builder.Build();
 
 app.UseSwagger();
 app.UseSwaggerUI();
 
 app.UseCors("AllowSpecificOrigin");
 
-app.UseExceptionHandler(errorApp =>
-{
-	errorApp.Run(async context =>
-	{
-		var exceptionHandlerPathFeature = context.Features.Get<Microsoft.AspNetCore.Diagnostics.IExceptionHandlerPathFeature>();
-		var env = context.RequestServices.GetService<IWebHostEnvironment>();
-		var errorCode = StatusCodes.Status500InternalServerError;
-		context.Response.StatusCode = errorCode;
-		context.Response.ContentType = "application/json";
-
-		var errorMessage = "An unexpected error occurred. Please try again later.";
-		
-		if (exceptionHandlerPathFeature?.Error is ArgumentException)
-		{
-			errorMessage = exceptionHandlerPathFeature.Error.Message;
-			errorCode = StatusCodes.Status400BadRequest;
-		}
-		else if (exceptionHandlerPathFeature?.Error is NotFoundException)
-		{
-			errorMessage = exceptionHandlerPathFeature.Error.Message;
-			errorCode = StatusCodes.Status404NotFound;
-		}
-		else if (exceptionHandlerPathFeature?.Error is ConflictException)
-		{
-			errorMessage = exceptionHandlerPathFeature.Error.Message;
-			errorCode = StatusCodes.Status409Conflict;
-		}
-
-		if (env.IsDevelopment())
-		{
-			// In development, return detailed error information
-			errorMessage += " dev info: " + exceptionHandlerPathFeature?.Error.InnerException;
-		}
-		await context.Response.WriteAsJsonAsync(new 
-		{ 
-			ErrorMessage = errorMessage, 
-			ErrorCode = errorCode 
-		});
-	});
-});
-
-
 // StatusCodePages Middleware
 app.UseStatusCodePages(context =>
 {
 	context.HttpContext.Response.ContentType = "application/json";
 	return context.HttpContext.Response.WriteAsJsonAsync(new { ErrorMessage = "An unexpected error occurred." });
 });
 
+app.UseRouting(); // This must come before UseAuthentication and UseAuthorization
+app.UseHttpsRedirection();
```

Comment:
This middleware can be tricky and can overwrite some cookies! Let's keep in mind that because of this, we can have some issues with the `authorization` header. No action needed here